catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/sharing/ShareableMetacardImpl.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.metacard.sharing;
+
+import com.google.common.collect.Sets;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.impl.types.SecurityAttributes;
+import ddf.catalog.data.types.Core;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/** A metacard that supports sharing. */
+public class ShareableMetacardImpl extends MetacardImpl {
+  private static final MetacardType TYPE = new ShareableMetacardType();
+
+  public ShareableMetacardImpl(MetacardType type) {
+    super(type);
+  }
+
+  public ShareableMetacardImpl(Metacard metacard) {
+    super(metacard);
+  }
+
+  public ShareableMetacardImpl(Metacard metacard, MetacardType type) {
+    super(metacard, type);
+  }
+
+  public ShareableMetacardImpl() {
+    super(TYPE);
+  }
+
+  public ShareableMetacardImpl(String id) {
+    this();
+    setId(id);
+  }
+
+  /**
+   * Check if a given metacard is a shareable metacard by checking the tags metacard attribute.
+   *
+   * @param metacard the metacard to check.
+   * @return true if the provided metacard is a shareable metacard, false otherwise.
+   */
+  public static boolean isShareableMetacard(Metacard metacard) {
+    return metacard != null
+        && metacard
+            .getMetacardType()
+            .getAttributeDescriptors()
+            .stream()
+            .map(AttributeDescriptor::getName)
+            .anyMatch(attr -> Objects.equals(SecurityAttributes.ACCESS_GROUPS, attr))
+        && metacard
+            .getMetacardType()
+            .getAttributeDescriptors()
+            .stream()
+            .map(AttributeDescriptor::getName)
+            .anyMatch(attr -> Objects.equals(SecurityAttributes.ACCESS_INDIVIDUALS, attr));
+  }
+
+  /** Wrap any metacard as a ShareableMetacardImpl. */
+  public static ShareableMetacardImpl from(Metacard metacard) {
```

Comment:
❓ We probably wouldn't need this wrapper, or maybe this entire class, if we incorporate sharing into catalog-core-api.

Reply:
I'm ok with incubating here for now. I'm glad we don't need to define any new attributes yet, though we'll probably have to revisit the concept of `metacard.owner` as metadata vs. permission at some point in the future.